### ISSUE TITLE
Call for Maintainers

### ISSUE THREAD

**Issue by brianreavis** [selectize/selectize.js#752]
I'm looking for new maintainers of Selectize. In the coming weeks/days, I'll be moving it to an organization account so other people can take care of it. The reasons behind the move (+ some other thoughts):
1. Selectize has been a personal project of mine – it's received no sponsorship and has been a "free time" project. I've been spread thin and it's been increasingly difficult to maintain as my available time has dwindled. I've been heads-down working [Natural Atlas](https://github.com/naturalatlas).
2. _jQuery plugins for UI components have always been a hack._ I'm really stoked by what [React](https://facebook.github.io/react/) is doing in this space. I highly recommend checking out @JedWatson's [**react-select**](https://github.com/JedWatson/react-select) project (even cooler given the fact Jed has contributed to Selectize in the past).
3. Lastly: **open-source burnout**. I work on other OSS projects and jQuery plugins seem to be the most pain afflicting. The burnout stems from a few factors:
   - Everyone has a different idea of how it should work. Different opinions and hard choices are great – but when the stream of people coming into Issues wanting you to drop everything else to make it _their_ way is incessant, it gets old.
   - Entitlement around new features, changes, and support. See bullet 1. 
   - Not enough ingenuity from users (i.e. the Stack Overflow effect). If a style is slightly off, change it for the project. If something minor is broken, fix it and make a pull request that makes obvious what the problem is, how it's solved, and ideally some tests to make sure it doesn't regress. _This is an open source project, not a company's product._
   - Front-end UI has a culture around un-tested code – after all, it's _hard_. Regardless, pull requests with tests are _extremely rare_, and most pull requests come in half-baked. By "half-baked" I mean they come with unintended consequences.
   - The readme is ignored: updating the dist files in a PR has never been the right thing to do.
   - Too many middle-of-the-night support requests.

What I'm mainly looking for in maintainer(s):
- Attention to detail. With the myriad of browser inconsistencies when it comes to focus handling combined with wide array of use patterns, it's easy to naively introduce new issues.

Where selectize needs some love:
- Bugfixes
- Finishing locking down the API.
- The "single" select mode that mimics a normal `<select>` could use some UX improvement. [**react-select**](https://github.com/JedWatson/react-select) is a great example of how it can be improved.

That said, thanks to [everyone](https://github.com/brianreavis/selectize.js/graphs/contributors) who's brought selectize to where it is now. There's lots I've learned and certainly much I could have done better. I still want selectize to succeed and will try to work on it from time to time, but it'll be less than I currently do. If you're interested in maintaining, please email me: brian@thirdroute.com Thanks!

**Comment by JedWatson** [comment]
@brianreavis I'm honoured. Thanks for all the work you've done on Selectize, you've created (imo) the best of the jQuery select controls and it was a huge inspiration for my work on react-select, especially with regards to the user interaction design.

I think many people (myself included, previously) underestimate how much work it is to get the _design_ of a component like this right, and you nailed it.

Good luck finding maintainers, hopefully some great people will step up and help from here!

**Comment by airwin** [comment]
Selectize.js is a great plugin, it helps me a lot, hope to have a good continuation! btw: I have turn to write react too and may also use @JedWatson 's react-select >.< . Thanks to all the open source contributors~!

**Comment by tannerlinsley** [comment]
I would love to contribute.  I'm run the entire front-end for Nozzle.io, and we exclusively use selectize.js for all of our needs (+ angular).

**Comment by gregblass** [comment]
Just wanted to say thanks Brian. Using it for a responsive website and its working great! Chosen wouldn't work on mobile at all, and didn't have some of the other features I needed baked in like adding choices (I ended up hacking chosen to do that, but it was nice to just have it available). Yesterday I spent a LONG time looking through this code base and thinking about how I could just move the selected items outside of the box. It took me way longer than it should have to come up with a solution (so far down a few rabbit holes, when they were just the wrong ones). Anyway I really got a sense for this code base, and damn - there is a lot going on! I agree with you...I think in a general sense that this is kind of ridiculous, and can see how jQuery may not quite be the best way to go about things. And I also read through every issue trying to find an insight on mine. Man, people get SO entitled! Good luck with your project. Cheers.

**Comment by gregblass** [comment]
Here's the responsive web app I finished up last night for a client using selectize magic! You can see I styled it a ton, and got the selected items to show up below - and working on mobile. Posting this to hopefully give you some sense of happiness to see your code being used. I wish I had the chops to contribute to the project. Thanks again!

![multi-select](https://cloud.githubusercontent.com/assets/279927/7090633/6ff44960-df72-11e4-8ce4-9c5a41f33dcc.jpg)

**Comment by barretts** [comment]
Thanks for all your work @brianreavis I've enjoyed using it these past few months. I'd love to contribute but I think you're right that the future is elsewhere. I've run into those focus issues in my own cross browser testing, nothing is ever simple is it? Though since I wasn't even able to get my own tests passing on Travis and haven't had the time to dig any deeper I don't think I could take the reins alone like you've been doing for so long. Anyhow best of luck on your future work, thanks again!

**Comment by mikemaccana** [comment]
@brianreavis another way the burden could be reduced is by dropping support for older browsers completely. Google are IE10+ right now.

**Comment by owais** [comment]
@brianreavis selectize is one of the most flexible and well designed frontend libs I've come across. You'd be surprised to see how I've used it in my projects. It never stops to be useful even for things it was not supposed to power. Thank you so much!

**Comment by thstarshine** [comment]
So what's the status of this project now please?
Is there anyone gonna taking care of this?
Selectize is a really nice plugin.

**Comment by tannerlinsley** [comment]
I'm thinking of putting together a rewrite of selectize.  Removing the jQuery dependency, and cleaning up some of the code.  Hopefully going to be more lightweight and have more opportunity for extensions. Another goal would be to separate the skinning and layout, and hopefully allow for some great framework extensions as well for angular, react, etc. Thoughts?

**Comment by nathanboktae** [comment]
> I'm thinking of putting together a rewrite of selectize. Removing the jQuery dependency,

I'd love to help with a version that removes the jQuery dependency. Now with evergreen browsers working directly with the DOM is just fine. Then having a library of thin wrappers around popular MVC frameworks would be great and help it's long term success.

**Comment by tomwjerry** [comment]
Any news about this rewrite stuff? Have you begun make a project yet? Selectize is like life and death for my project. If none are gonna make it, I will. I am not a coding expert and I know little about coding without jQuery. What I would to have is selection of multiple entries and creation of new entries on the fly.

Hope that the new selectize project can get started soon!

**Comment by u01jmg3** [comment]
I would be happy to help with maintaining this library to the high standard you have already put in place. I've used it extensively for a current project and have several bug fixes ready and tested but will wait to PR until the queue goes down -- hopefully I can help with that.

**Comment by tomwjerry** [comment]
I still think we should need to make an library independent selectize thing. I do not think he will accept any fixes.

**Comment by drownbes** [comment]
So... Project is dead? What are the best alternatives to it?

**Comment by philfreo** [comment]
This repo may not be maintained, but the project itself isn't dead and having recently researched alternatives, I still think this is the best codebase / project like it.

**Comment by tannerlinsley** [comment]
If someone wants to port it over to the github.com/selectize organization,
I can get you access.  Just let me know.

**Comment by tomwjerry** [comment]
The question is if the project should be rewritten from scratch or if just the code should be changed to remove the jQuery dependency. Should not be too hard to just change the code so that jQuery is not used.

**Comment by chipx86** [comment]
Rewriting from scratch essentially turns this into a new project. Perhaps a similar/compatible API, but it means possibly introducing some breaking bugs or impacting any custom plugins that a project may be using.

If there were two forks of this, one rewritten from scratch and one maintaining/fixing bugs in the existing codebase, I know for sure that we'd choose the latter, as we couldn't risk the potential negative impacts of a brand new codebase.

I think if it were rewritten from scratch, it'd be best to also rename the project in the process.

Whoever does take this on, please ensure that if the jQuery dependency is dropped, custom plugins won't be negatively impacted, or at the very least, any breaking changes are well-documented.

Same with the API in general. `$('#myel').selectize()` should still be able to work if jQuery is already on the page, for instance, or it may impact a lot of code out there.

**Comment by joallard** [comment]
I've been thinking about breaking some things down into smaller modules/breaking up the API a bit (for instance to isolate the "fake input" functionality), pondering this over the past year but never making a move. I'd be down to pitch in with maintaining/refactoring/participating/triaging/whatever, with the possibility of removing jQuery for a Major bump. I for one am committed to a clean API and an extensible plugin.

Have some people been selected or has there been any progress? I would've liked to make some PRs, but the queue is quite long.

**Comment by antitoxic** [comment]
Hey @brianreavis, is there a way to check whether anyone is chosen as a new maintainer? 

Also I can't really see the github ["Network" tab](https://github.com/brianreavis/selectize.js/network) which shows how forks are progressing because there are too many forks. So basically I can't see whether anyone took over.

**Comment by tomwjerry** [comment]
None took over this so far. Best luck is on github.com/selectize

**Comment by joallard** [comment]
@tomelssjo @tannerlinsley selectize/selectize doesn't contain any of the Git history, it's just a crude directory copy. If we do it right I'd be happy to pitch in and do some Git-fu with the PRs here, even better if we had @brianreavis' benediction.

**Comment by tannerlinsley** [comment]
For sure. A transfer would be optimal with a rewrite of the contributors.md.
Next best thing to that is some git fu like you said. I am happy to give
access to the repo to anyone who wants to tackle it.

**Comment by joallard** [comment]
So @tannerlinsley and I have set up an organization fork at [selectize/selectize](https://github.com/selectize/selectize) to organize efforts to put this back on the rails. If you'd like to offer your help, chime in at selectize/selectize#4 or open an issue to talk about it. Or if there are PRs you'd like to resubmit/port over, go ahead. Hope to see you there!

**Comment by brianreavis** [comment]
Thanks for spearheading this @joallard and again I'd like to apologize for the lack of responsiveness. Trying to juggle a lot these days. Mind adding me as an owner to the selectize organization? I'll transfer ownership over there. To answer the question of "have maintainers been selected"? Sadly no. Only a handful (can count on one hand) have come forward and unfortunately none have been a great match. It's going to take some strict direction and willingness to be in maintenance 80% of the time. It's a lot of work. It's sounds like you've got the right idea though, so I'm happy to do whatever I can to at least get it moving again.

**Comment by joallard** [comment]
No worries, I'm really glad we have your benediction Brian! Your guidance will be greatly appreciated and some help whenever/if you feel like it. 

I agree that it's going to take a strong vision to go forward both _in_ code and _around it_ in organization, which I'm happy to provide. I come from a Ruby background, I'm used to delving into Rails and other good gems' source code, so I think I have what it takes in terms of clean code, clean API, encapsulation and modularity.

I think we can succeed if we have multiple solid maintainers that are on the same page (ideally—as we know, it's easy to get swamped up when one is the sole maintainer) and help with things such as issue triage, patch review, etc. I definitely have a few ideas around in my head on that front.

(I don't know what you meant by "transferring ownership"; I cloned the master branch over there to keep all history, I hope we get to keep current issues and PRs for reference?)

(@tannerlinsley, would you please add him as org owner?)

**Comment by brianreavis** [comment]
I see @tannerlinsley added me to the repo but not the organization it seems (try https://github.com/orgs/selectize/people). After that's fixed, I'll rename your clone to `selectize-temp` or something and move this repo over (issues and pull requests included). Then we can copy-paste your couple issues you have into the complete one.

**Comment by tannerlinsley** [comment]
@brianreavis I added you as an owner so you should be able to admin the the
transfer of the repo. That'll be great to keep all the PRs and issues.

**Comment by brianreavis** [comment]
Thanks @tannerlinsley. Alright, the move is complete... it should be all good to go.

**Comment by guoyunhe** [comment]
I think selectize is much better than select2 for usability. Hope this project can go further. I am planing to add Bootstrap 3 SASS and make a pull request.

**Comment by joallard** [comment]
A few news on this, I've volunteered since the start of the year to help with maintaining. I do my best for triaging bugs and PRs and keeping on top of notifications, but this is a large project and I'm still not fully up to speed right now. If some more people with the proper experience want to pitch in maintaining/triaging/bug-fixing/developing, I'd be glad to organize that. But for the moment, it's just me in my spare time trying to figure out boring stuff like a reproduceable dev environment, remote plugin repos, triaging priority, etc. in between managing sometimes very general questions posted as issues and various requests. That's your news flash for now!

**Comment by ppawel** [comment]
AWESOME!

**Comment by joallard** [comment]
Here we are folks, after picking up where @brianreavis left off, I find myself unable to fulfill my maintaining duties anymore.

I first I thought, it's okay if I'm not there to fix everything, we just need to have a solid process to report issues and accept PRs. Turns out, it's really hard to do that.

I introduced new guidelines for issues and PRs. Having a simple process of Open link and Reproduce for bugs makes it easier to triage bugs. However, even with a multitude of warnings and guides, there is still only a minority of reporters who do that, and treat the *Issue tracker*—of which we have now 408 because of that very reason—as a place to put things that are questions or "here, debug my code" without paying attention to the guidelines. I do want to thank the people who put care in filing those to make them easy to deal though. I wish there were more of you.

The PRs are another type of hard, where you never really know if something will break some other thing, whether you'll introduce a regression or just not completely fix the problem. Add that to other hurdles such as: the PR just ignoring tests, changing `dist/`, having no peer review, scope creep, and uncertain code quality. Shepherding PRs is hard.

I unsubscribed from the notifications. In between looking for work last year (couldn't afford spending time that wasn't looking for work) and getting work this year (don't have time anymore), it often comes down to that it's pretty damn hard to work on a library that's as complex and as popular as this without being paid.

Add that to the fact that I'm out of the projects where I used Selectize for now, so it's hard to organically add stuff and patch things (scratching my own itches) in the library. That's another big one.

So in order for the project to move forward, I'll have to allow the project to seek a new maintainer but more ideally, seek **a team of maintainers**. Cause doing this as a one-person job (person who need to pay their rent *some way*) is not really setting us up for success and produces that kind of burnout.

**Comment by brianreavis** [comment]
Hey @joallard, thanks so much for all you've done. I personally appreciate it all, and I'm sure others do too. It's not easy work and you've done an awesome job at it.

**Comment by jbrooksuk** [comment]
Put my name down :)

**Comment by jblacker** [comment]
I'm can't say I have much time, but I've been using this in a few of my projects and found it immensely helpful.  I don't want it to die, so I'd be happy helping out wherever/whenever I can.

**Comment by siegy22** [comment]
@brianreavis If you missed it, @jbrooksuk and @jblacker could help this project out. 💚

**Comment by yrshaikh** [comment]
Hey, I know I am late to the party. But I can help too 👍

**Comment by coderberry** [comment]
Do you have an open collective for this project? I'd like to invite you to set this up on [Code Sponsor](https://codesponsor.io) to help bring in funding if that would help. We use it in our app and love it!

**Comment by Pictor13** [comment]
I am willing to help, as well, with reviewing, maintenance of issues, eventually bug fixing and minor implementations.
I think a lot of issues asking for support are not relevant anymore, as being too old or being solved, and could be closed (users can reopen and keep issues active, if needed).

**Comment by anx-ckreuzberger** [comment]
selectize js is a great library, and I would love to see getting a new release, e.g., 0.13.0, with those bugfixes etc...! Don't get burned out, don't get stressed by the number of issues, features, pull requests that are "asked" for. 

Please keep up the good work, consider setting up some ways to sponsor the devs (e.g., patreon, or codesponsor, or opencollective, etc..). I am sure many would be willing to donate to get some work done.

**Comment by JefferyHus** [comment]
Count me in to help anytime. First time I use this library

**Comment by Kovah** [comment]
Selectize is kind of dead now, isn't it? :(

Edit: @joallard @brianreavis  mind responding here? There are so many people offering help. Or maybe just archive the project if it's already abandoned?

**Comment by mikemaccana** [comment]
+1 for archiving the project. **Selectise was awesome for it's day** and I'm very grateful to Brian and co. If you were making something now you'd have a Svelte or React or Vue component (that was probably inspired by Selectise!).

**Comment by jblacker** [comment]
+1 for archiving as well at this point.  Back when I commented on this thread in 2017 I was all for helping to maintain, however front end technology & frameworks have progressed so quickly that this library isn't really as useful as it used to be.  As @mikemaccana mentioned most people would use an Angular, React, Vue, or another framework component, or even a web component with some babel magic to make it compatible with all browsers instead of a jQuery-based tool.

This library was incredible in the past and I loved using it in my projects, but it's at the end of its useful life at this point.  I see lots of components out there that look very similar to selectize and even use the word "selectize" as a theming option; it has certainly left its mark on the world.

**Comment by devkral** [comment]
Can I restart the development? I still see a future for this framework.
Especially if getting rid of jquery and also restarting the development of sifter.
One main selling point is the good mobile support I missed with select2.

But expect no wonders I have also not so much time and it will be a spare time project.

**Comment by devkral** [comment]
my name on npm is also devkral

**Comment by JefferyHus** [comment]
Ok, whoever is up and ready to maintain this repository, to like this comment, comment after with your DISCORD `username`. Let us get this beauty back to life.

**Comment by scarroll32** [comment]
129 open pull requests ... did anyone finally agree to become a maintainer?

**Comment by JefferyHus** [comment]
We have not heard back from the Author yet

**Comment by stephanvierkant** [comment]
Ping @joallard @brianreavis

**Comment by devkral** [comment]
I switched meanwhile to choose.js and added support for select boxes (not merged yet, still some minor issues)

**Comment by joallard** [comment]
Okay, looks like meanwhile I've gotten admin rights from Brian to be able to nominate maintainers.

This is all a bit stale, so if you're up for it, please make your case. The ideal volunteer is someone who uses the library and knows it well, has participated in issues/PRs before, and who can help move this project forward with the input of the community. 

The job is pretty much to review stuff, help people write good code, and best case, scratch their own itches because they significantly use the project. (last point optional)

If there's anything I've learned in open source: it takes a lot of patience, people can be from ungrateful to lovely — sometimes running into an annoying bug for hours can be really tiring too! — and it's real work.

Like I said, ideally, this would look like a team of 2-3.

I'm able to put a tiny bit of help with really meta things, but I can't give actual work time to this, I'm sorry. That said, if we're able to assemble something together so work is supported/funded, that'd be fantastic.

Hopefully this is a new start for this old project ;) — and the people who appreciate it.

**Comment by databyte** [comment]
Hey @joallard - I think at a minimum we should have a couple people help maintain repo health by managing travis, package.json and security vulnerabilities. Even if I'm not working on feature requests, I'd be great to maintain a healthy repo overall.

As a contributor on a couple other packages, it's a lot of work to drum up support so lets use the recent activity around security vulnerabilities for library dependencies to pull in a few folks. If we could define a simple minor version update, then we can focus the next set of PRs and cleanup the Issues list.

I recently used selectize on a project because they weren't using a client-side framework (e.g. React, Vue, etc) and it met the needs more so than other drop-down selectors. As an active and recent user, I don't see any reason why we couldn't pull together a small group of folks in Github discussions or Discord to cleanup and release a new version after a good 8-12 hours of work.

We may want to call bankruptcy on Issues older than 2 years then triage the rest.

I'm game if anyone else wants to sink a couple days into cleaning this up!

**Comment by oyejorge** [comment]
To fork or not to fork?

This project is awesome and it would be a shame to let it fade away. I personally think development should remove dependence on jQuery and aim for a framework agnostic design. I've completed removing jQuery, made sure tests pass, updated some documentation, etc [over here](https://github.com/orchidjs/tom-select) and I'm wondering if I should continue independently or merge into selectize? @joallard @brianreavis

**Comment by scarroll32** [comment]
Wow @oyejorge you have a serious amount of work in your fork. I think removing jQuery is a great idea, your fork should be accepted back here as a major version bump.

**Comment by joallard** [comment]
Hey folks, thanks for your interest here. I thought I had asked in my previous post, but alas I did not: people interested in maintenance, would you send me an email at `atob("am9uYXRoYW4uYWxsYXJkQGhleS5jb20=")`? I'm hoping we'll be able to talk it out from there. — cc @databyte @oyejorge and others interested

Alas, folks waiting for PRs will have to wait on this blocker to be resolved. Thanks for holding on tight, I'm sure it feels pretty ungrateful not to respond to PRs, but I'm hoping we'll be able to reuse your work soon enough when we're able to.

**Comment by Pictor13** [comment]
Ehm... any news?
Was there any meeting or decision with the mentioned volunteers?

I see there is @risadams that made several commits on the master branch and added himself for the lib's Copyright starting from 2020, so I assume somebody gave him write permissions or handed the repo to him, even though he never wrote in this thread to propose himself.

Who  gave the permissions, @brianreavis or @joallard?
Was there some relevant decision that was not reported here yet?

It's good though, to see some refreshing work being done on Selectize. 👍

**Comment by databyte** [comment]
Speaking of which, I was just talking to @risadams and @joallard over keybase about maintainers. Join the conversation - there's a general chat room on the open team `#selectize`.

We can move our conversation to the public channel there.

It's suggested we should get roughly 4-5 contributors then refresh the Project list, cleanup open Issues and work through the PRs. @joallard did some work already on getting CI cleaned up, merged a couple simple PRs and then added bootstrap 4 support. Lots more to get done.

We only just started chatting today so it's still early.

via app: keybase://team-page/selectize
via web: https://keybase.io/team/selectize

**Comment by joallard** [comment]
@Pictor13 Confirming, I did. @risadams and @databyte responded to my previous comment, and I really didn't have the time to dedicate to this, so I took a bit of time to set up a maintenance team. I'm hoping this is some fresh new energy into this project!

**Comment by Pictor13** [comment]
Glad to hear that!
And sorry guys if I rushed things 😅
I'll follow and try to contribute as well when I can.

Thanks everybody for your past, present and future work 🙂

**Comment by risadams** [comment]
@Pictor13 thanks for jumping in so quickly!

**Comment by anandphulwani** [comment]
@joallard I would like to put my name in the list.
